sqlglot/dialects.py

```diff
@@ -411,6 +484,62 @@ def _time_to_unix(self, expression):
     def _unix_to_time(self, expression):
         return f"FROM_UNIXTIME({self.sql(expression, 'this')})"
 
+    def _ts_or_ds_or_di_to_date_str_sql(self, expression):
+        this = self.sql(expression, "this")
+        return Hive.TS_OR_DS_OR_DI_TO_DATE_STR_EXPRESSION.format(this=this)
+
+    def _ts_or_ds_or_di_to_di_sql(self, expression):
+        this = self.sql(expression, "this")
+        return Hive.TS_OR_DS_OR_DI_TO_DI_EXPRESSION.format(this=this)
+
+    def _ts_or_ds_or_di_to_date_sql(self, expression):
+        this = self.sql(expression, "this")
+        return Hive.TS_OR_DS_OR_DI_TO_DATE_EXPRESSION.format(this=this)
+
+    def _di_add_sql(self, expression):
+        this = self.sql(expression, "this")
+        e = self.sql(expression, "expression")
+        unit = self.sql(expression, "unit") or "'DAY'"
+
+        date = Hive.TS_OR_DS_OR_DI_TO_DATE_EXPRESSION.format(this=this)
+
+        if unit == "'DAY'":
+            add_expression = f"DATE_ADD({date}, {e})"
+        elif unit == "'MONTH'":
+            add_expression = f"ADD_MONTHS({date}, {e})"
+        elif unit == "'YEAR'":
+            add_expression = f"ADD_MONTHS({date}, {e} * 12)"
+        else:
+            raise NotImplementedError(f"Unit not implemented for Spark Di Add: {unit}")
+
+        return f"CAST({add_expression} as int)"
+
+    def _ts_or_ds_or_di_date_diff_sql(self, expression):
+        third_arg = self.sql(expression, "second_date")
+        if third_arg:
+            first_date_value = self.sql(expression, "first_date")
+            second_date_value = third_arg
+            unit = self.sql(expression, "unit")
```

Comment:
this will be easier if you do expression.text("unit") then you don't hae to check the quotes in the string